components/user_access/app/services/user_access/user_registration_service.rb

```diff
@@ -0,0 +1,56 @@
+module UserAccess
+  class UserRegistrationService
+    attr_reader :login, :password, :password_confirmation, :email, :first_name, :last_name
+
+    def initialize(attributes)
+      @login = attributes[:login]
+      @password = attributes[:password]
+      @password_confirmation = attributes[:password_confirmation]
+      @email = attributes[:email]
+      @first_name = attributes[:first_name]
+      @last_name = attributes[:last_name]
+    end
+
+    def call
+      return false unless user_registration.valid?
+
+      ActiveSupport::Notifications.publish(
```

Comment:
@pmcruz90 check this event and the subscriber, and how they interact.

You can check the README of the repo to understand the events and business logic.